### ISSUE TITLE
State not being serialized correctly?

### ISSUE THREAD

**Issue by ianks** [zalmoxisus/redux-devtools-extension#81]
After upgrading to redux 3.3.1, the extension shows all of my state ass looking something like this:

`$jsan:"$.actionsById.5.action.payload.props.children[0]._owner._renderedComponent._renderedComponent._renderedChildren[".0"]._renderedComponent._instance.state.storeState"`

Any idea why this is happening?

**Comment by zalmoxisus** [comment]
Our examples also [use redux@3.3.1](https://github.com/zalmoxisus/redux-devtools-extension/blob/master/examples/counter/package.json#L23). You can [give it a try](http://zalmoxisus.github.io/redux-devtools-extension/examples/counter/). Is it the same there?

Could you please provide more details? In which monitor do you see this?

**Comment by ianks** [comment]
That example works fine.

The issue is reproducible in all monitors.

**Comment by zalmoxisus** [comment]
Thanks for the details. For some reason [jsan](https://github.com/kolodny/jsan) fails to serialize the object.

You could of course disable the serialization in the extension's settings, and that should solve the issue, but it would be welcome to get a test to replicate this and be able to fix `jsan`.

Could you please paste the plain object which should be serialized from `store.liftedStore.getState()`. The easiest way to do this is to expose `store` object globally and run that in the console and paste the result here. Then we can replicate it with `jsan.parse(jsan.stringify(obj))`, which should return the object back.

**Comment by ianks** [comment]
Here is the raw object copy and pasted from dev tools.
```
Object
@@reduxReactRouter/routerStateSelector
:
routerStateSelector(state)
dispatch
:
(action)
getState
:
getState()
history
:
Object
liftedStore
:
Object
replaceReducer
:
replaceReducer(n)
subscribe
:
subscribe(listener)
__proto__
:
Object
```

After running `jsan.parse(jsan.stringify(obj))`, the resulting object is missing a few keys:

```
Object {liftedStore: Object, history: Object}
history
:
Object
liftedStore
:
Object
__proto__
:
Object
```

**Comment by zalmoxisus** [comment]
@kolodny, any idea what could be the problem here? Is it related to `jsan`?

**Comment by kolodny** [comment]
jsan works in one of two modes, there's the slower fuller featured mode which walks the object and marks functions, self references and other non-JSON types, the other mode is the fast mode which it tries first be default. Doing `jsan.parse(jsan.stringify(obj))` does the fast way first and if it doesn't fail, which it looks like it didn't, happily returns that. That drops functions and any other non-JSON type.

It doesn't sound like that's the issue here. I think I need a little more info to replicate. @ianks Can you provide a working (non-working) example?

**Comment by zalmoxisus** [comment]
@ianks, you provided the store object, we need the object which is the result of `store.liftedStore.getState()`. It should be something similar to what you provided in the first message:
`$jsan:"$.actionsById.5.action.payload.props.children[0]._owner._renderedComponent._renderedComponent._renderedChildren[".0"]._renderedComponent._instance.state.storeState"`

**Comment by zalmoxisus** [comment]
@ianks any updates on this? Could you please also try to disable serialization on the option page (in case you don't use Immutable.js and have no circular references) and see weather it helps.

**Comment by ianks** [comment]
I upgraded to the new version of redux. This seems to have fixed my issue.

**Comment by ianks** [comment]
So I re-opened this issue, but this time I have more information. This bug only appears if I open the Developer Tools console *after* the page has loaded. If Developer Tools is open as I load the page, every thing works perfectly.

**Comment by zalmoxisus** [comment]
@ianks, if you have the monitor already opened, we send only the changes one by one, thus in first case we send the whole `store.liftedStore.getState()`. It would be useful to paste here the object returned from that operation (as [described above](https://github.com/zalmoxisus/redux-devtools-extension/issues/81#issuecomment-203039027)), so we could check whether the problem is in serialization.

**Comment by ianks** [comment]
store.liftedStore.getState() looks entirely normal. All of the actions look OK. All of the state is correct.

```
Object {monitorState: Object, actionsById: Object, nextActionId: 15, stagedActionIds: Array[15], skippedActionIds: Array[0]…}
```

Attempting to call `jsan.stringify(store.liftedStore.getState())` freezes my dev tools console.

**Comment by zalmoxisus** [comment]
@ianks, thanks for the information. Yes, the object structure is correct, but I still don't know what is inside.

Could you please provide also `store.liftedStore.getState().computedStates` and `store.liftedStore.getState().actionsById`? Optionally you can try to stringify these objects as well.

**Comment by ianks** [comment]
I do not feel comfortable posting the whole state here, but I can confirm this:

`store.liftedStore.getState().computedStates` returns the state, and it is entirely correct.

`store.liftedStore.getState().actionsById` contains a circular reference which causes jsan.serialize to freeze.

**Comment by zalmoxisus** [comment]
@ianks, thanks a lot! Knowing that a circular reference is the reason is sufficient. I'm able to reproduce it [here](http://zalmoxisus.github.io/redux-devtools-extension/examples/router/). The problem was that we have an older version of `jsan` in `remotedev-app`, so we were serializing with one version and parsing with another one.

Fixed in `v1.0.0.20`, please give it a try when you have time.
